Neos.Flow/Classes/Mvc/DispatchComponent.php

```diff
@@ -75,90 +37,22 @@ public function __construct(array $options = [])
      */
     public function handle(ComponentContext $componentContext)
     {
-        $componentContext = $this->prepareActionRequest($componentContext);
         $actionRequest = $componentContext->getParameter(DispatchComponent::class, 'actionRequest');
-        $this->setDefaultControllerAndActionNameIfNoneSpecified($actionRequest);
         $actionResponse = $this->prepareActionResponse($componentContext->getHttpResponse());
-        $this->dispatcher->dispatch($actionRequest, $actionResponse);
+        try {
+            $this->dispatcher->dispatch($actionRequest, $actionResponse);
+        } catch (AuthenticationRequiredException $exception) {
```

Comment:
After rethinking, given that the ActionResponse can carry component context parameters, I guess we should move this catch back to the dispatcher and set the respective parameter there.

We just first need to address the merging of ActionResponse to http response and component context. which is still a TODO.